pym/gentoolkit/eclean/search.py

```diff
@@ -332,6 +332,19 @@ def _non_destructive(
             deprecated.update(_deprecated)
         return pkgs, deprecated
 
+    def git_check(self, distdir):
+        cpvs = [
+            i.split("/")[1][:-5]
+            for i in set(self.vardb.cpv_all())
+            if "live" in self.vardb.aux_get(i, ["PROPERTIES"])
+        ]
+        git_src = os.path.join(distdir, "git3-src")
+        gitdir = dict(map(lambda i: (i.split("_")[1][:-4], i), os.listdir(git_src)))
+        deprecated_git = [
+            os.path.join(git_src, gitdir[i]) for i in gitdir.keys() if i not in cpvs
+        ]
+        return deprecated_git
```

Comment:
Please add a comment to explain what this is doing. It's very hard to read.

I suspect you are matching directory names in git3-src to ebuild package names. That seems incorrect to me: git-r3.eclass does not use the ebuild package name to calculate the `repo_name` variable.

Reply:
Yes @floppym I was doing that now I'm using the env file for the path.